File: src/main/java/org/janelia/saalfeldlab/paintera/data/n5/ReflectionHelpers.java
package org.janelia.saalfeldlab.paintera.data.n5;

import java.lang.reflect.Field;

public class ReflectionHelpers
{

	public static Field searchForField(
			final Class<?> startClass,
			final String name) throws NoSuchFieldException
	{

		try
		{
			final Field field = startClass.getDeclaredField(name);
			field.setAccessible(true);
			return field;
		} catch (final NoSuchFieldException e)
		{
			final Class<?> superClass = startClass.getSuperclass();
			if (superClass == null)
				throw e;
			return searchForField(superClass, name);
		}
	}
}

File: src/main/kotlin/org/janelia/saalfeldlab/paintera/data/n5/ReflectionHelpers.kt
package org.janelia.saalfeldlab.paintera.data.n5

import java.lang.reflect.Field

class ReflectionHelpers {

    companion object {
        @Throws(NoSuchFieldException::class)
        @JvmStatic
        fun searchForField(
            startClass: Class<*>,
            name: String
        ): Field {

            return try {
                startClass.getDeclaredField(name).also { it.isAccessible = true }
            } catch (e: NoSuchFieldException) {
                val superClass = startClass.superclass ?: throw e
                searchForField(superClass, name)
            }

        }
    }
}
